src/multi_vector_simulator/E0_evaluation.py

```diff
@@ -197,12 +200,22 @@ def evaluate_dict(dict_values, results_main, results_meta):
 
     # Tests and checks
     logging.info("Running validity checks.")
-    E4.minimal_constraint_test(dict_values, MINIMAL_RENEWABLE_FACTOR, RENEWABLE_FACTOR)
-    E4.minimal_constraint_test(
-        dict_values, MINIMAL_DEGREE_OF_AUTONOMY, DEGREE_OF_AUTONOMY
-    )
-    E4.maximum_emissions_test(dict_values)
-    E4.net_zero_energy_constraint_test(dict_values)
+
+    if MINIMAL_RENEWABLE_FACTOR in dict_values[CONSTRAINTS]:
+        E4.minimal_constraint_test(
+            dict_values, MINIMAL_RENEWABLE_FACTOR, RENEWABLE_FACTOR
+        )
+
+    if MINIMAL_DEGREE_OF_AUTONOMY in dict_values[CONSTRAINTS]:
+        E4.minimal_constraint_test(
+            dict_values, MINIMAL_DEGREE_OF_AUTONOMY, DEGREE_OF_AUTONOMY
+        )
+    if MAXIMUM_EMISSIONS in dict_values[CONSTRAINTS]:
+        E4.maximum_emissions_test(dict_values)
+
+    if NET_ZERO_ENERGY in dict_values[CONSTRAINTS]:
+        E4.net_zero_energy_constraint_test(dict_values)
+
```

Comment:
This makes sense as long as not all constraints are expected to be in the inputs. Fixing #780 would ensure that the parameters are always there, then the loops would not be necessary...

Reply:
the loops ? you mean if statements?

Reply:
Yes, thats what I ment. But you are right, the validity checks should only be applied when the constraints are added (eventhough in theory they should all return positive even if the constraints are turned off).